### PR TITLE
lncli: add deletepayments command

### PR DESCRIPTION
**Pull request by guggero**
Depends on https://github.com/lightningnetwork/lnd/pull/5660.

To give the CLI user the option to delete a single or multiple payments
in one command, we expose the DeletePayment and DeleteAllPayments
RPCs to the command line as well. Due to the similarity of the two RPCs
we combine them into a single command with multiple flags.

To make the command a bit more safe to run without arguments, we
consciously switch the logic of the RPC flag "failed_payments_only"
which is false by default into a "--include_non_failed" in the CLI which
is false by thefault. So a user running the command without knowing what
they are doing are only deleting failed payments by default, not all of
the payments.

We also do some re-grouping and code movement to consolidate all payment
related commands into the same file.

### CODE REVIEW

**Review by carlaKC — APPROVED** [review]
tACK for the last 4 commits, nice refactoring in this as well!

Only question I have here is whether we want to have an "are you sure + this might take a while" prompt for when we delete all payments?

File: cmd/lncli/cmd_payments.go [hunk 1]
```diff
@@ -1394,6 +1394,123 @@ func buildRoute(ctx *cli.Context) error {
...
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name: "payment_hash",
+			Usage: "delete a specific payment identified by its " +
+				"payment hash",
+		},
+		cli.BoolFlag{
+			Name: "failed_htlcs_only",
+			Usage: "only delete failed HTLCs from payments, not " +
+				"not the payment itself",
```
Comment (yyforyongyu) [inline cmd/lncli/cmd_payments.go:1432]: typo: extra `not`

**Review by yyforyongyu — COMMENTED** [review]
LGMT, just a typo and pending a release note.

**Comment by Roasbeef** [comment]
Lol I totally missed this when I made https://github.com/lightningnetwork/lnd/pull/5778

File: cmd/lncli/cmd_payments.go [hunk 2]
```diff
@@ -1394,6 +1394,123 @@ func buildRoute(ctx *cli.Context) error {
...
+	the internal bbolt database. But that disk space is only reclaimed after
+	compacting the database. Users might want to turn on auto compaction
+	(db.bolt.auto-compact=true in the config file or --db.bolt.auto-compact
+	as a command line flag) and restart lnd after deleting a large number of
+	payments to see a reduction in the file size of the channel.db file.
+	`,
+	Action: actionDecorator(deletePayments),
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name: "payment_hash",
```
Comment (Roasbeef) [inline cmd/lncli/cmd_payments.go:1425]: How about requiring the `-all` flag like I did in (https://github.com/lightningnetwork/lnd/pull/5778) to make this harder to use incorrectly? So just typing `lncli deletepayments` returns with the help, and forces you to understand what's going on and pick an option.

**Review by Roasbeef — CHANGES_REQUESTED** [review]

File: cmd/lncli/cmd_payments.go (hunk 2 above)
Comment (guggero) [reply cmd/lncli/cmd_payments.go:1432]: Yeah, good idea. Done :+1:

**Review by Roasbeef — APPROVED** [review]
LGTM 🌂

**Comment by Roasbeef** [comment]
Just needs the release notes, and this is g2g

**Pull request merged**